src/renderer/apiFetch.ts

```diff
@@ -0,0 +1,79 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import { apiActions } from "readium-desktop/common/redux/actions";
+import { TApiMethod, TApiMethodName } from "readium-desktop/main/api/api.type";
+import { TMethodApi, TModuleApi } from "readium-desktop/main/di";
+import { diRendererGet } from "readium-desktop/renderer/di";
+import { ApiLastSuccess } from "readium-desktop/renderer/redux/states/api";
+import { ReturnPromiseType } from "readium-desktop/typings/pierre";
+import { Unsubscribe } from "redux";
+import * as uuid from "uuid";
+
+export async function apiFetch<T extends TApiMethodName>(path: T, ...requestData: Parameters<TApiMethod[T]>) {
+    return new Promise<ReturnPromiseType<TApiMethod[T]>>((resolve, reject) => {
+        const store = diRendererGet("store");
+        const requestId = uuid.v4();
+        const moduleId = path.split("/")[0] as TModuleApi;
+        const methodId = path.split("/")[1] as TMethodApi;
+        let lastSuccess: ApiLastSuccess | undefined;
+        let storeUnsubscribe: Unsubscribe| undefined;
+        let timeout: number | undefined;
```

Comment:
1) I think that the variable `timeout` should be renamed to `timeoutId` (just like in https://github.com/readium/readium-desktop/blob/develop/src/utils/debounce.ts ) because otherwise this gives the reader the impression that `timeout` is a `number` of seconds (when in fact it is a unique "identifier").
2) To avoid hard-coding the TypeScript type `number` vs. `NodeJS.Timer`, would it be better to use `ReturnType<typeof setTimeout>` instead?

Reply:
https://github.com/readium/readium-desktop/pull/696/commits/9c17d0bf9740edb8a7a58d28befef85ef0d33cc8

Reply:
626b2d8

returnType doesn't works on compilation